test/parser/statement/Misc.test.js

```diff
@@ -0,0 +1,52 @@
+const brs = require("brs");
+
+describe("parser", () => {
+    let parser;
+
+    beforeEach(() => {
+        parser = new brs.parser.Parser();
+    });
+
+    describe("`end` keyword", () => {
+        it('does not produce errors', () => {
+            let { tokens } = brs.lexer.Lexer.scan(`
+                sub Main()
+                    end
+                end sub
+            `);
+            let { statements, errors } = parser.parse(tokens);
+            expect(errors.length).toEqual(0);
+            expect(statements).toMatchSnapshot();
+        });
+        it('can be used as a property name on objects', () => {
+            let { tokens } = brs.lexer.Lexer.scan(`
+                sub Main()
+                    person = {
+                        end: true
+                    }
+                end sub
+            `);
+            let { statements, errors } = parser.parse(tokens);
+            expect(errors.length).toEqual(0);
+            expect(statements).toMatchSnapshot();
+        });
+
+        it('is not allowed as a standalone variable', () => {
+            let { tokens } = brs.lexer.Lexer.scan(`
+                sub Main()
+                    end = true
+                end sub
+            `);
+            let { statements, errors } = parser.parse(tokens);
+            expect(errors.length).toEqual(1);
+            expect(errors[0].location.start.line).toEqual(3)
+            //TODO - re-enable this test once the `end` identifier location bug is fixed
+            // //specifically check for the error location, because the identifier location was wrong in the past
+            // expect(errors[0].location).toEqual({
```

Comment:
My change introduced an issue in the lexer that is incorrectly computing the position of the `end` token with my new change. It's off by 1 character (i.e. says char 8 but should be char 7). I tried to figure it out, but couldn't. 

The error is close enough to the correct position, and subsequent tokens get the proper location, so I don't think this is a show stopper. However, it would be great if you could take a look at this once the PR gets merged, and let me know what is wrong.

Reply:
Interesting!  I can't seem to reproduce that with simple files:

```brightscript
sub main()
    end
end sub
```

Produces this token for `"end"`:

```json
{
  "kind": 28,
  "text": "end",
  "isReserved": false,
  "location": {
    "start": {
      "line": 2,
      "column": 4
    },
    "end": {
      "line": 2,
      "column": 7
    },
    "file": "/path/to/the/file.brs"
  }
}
```

Looks okay to me?

Reply:
@TwitchBronBron do you happen to have a file that'd repro the off-by-one you mentioned above?  I'd like to get a feel for what kinds of situations are affected before I merge this.

Reply:
The error shows up when you use an assignment with end. Like this: 
```'master branch
sub main()
   end = 12
end sub
```
![image](https://user-images.githubusercontent.com/2544493/54881045-146c3080-4e22-11e9-9599-010a947bdb15.png)

I added a comment in this PR to a commented out test that shows the issue.

Reply:
Finally got some time to figure this out!  Turns out there was a small bug in column tracking involving the `end` string: we attempt to read the next word to see if it's a multiword `end if`, `end while` etc. and reset the value of `current` (the index into the string representation of the file we're lexing), but not the value of `column` (the index into the specific line in that file).


This should fix things:

```diff
diff --git a/src/lexer/Lexer.ts b/src/lexer/Lexer.ts
index 5799f81..817ccb7 100644
--- a/src/lexer/Lexer.ts
+++ b/src/lexer/Lexer.ts
@@ -503,7 +503,10 @@ export class Lexer {
 
             // some identifiers can be split into two words, so check the "next" word and see what we get
             if ((text === "end" || text === "else" || text === "exit" || text === "for") && (peek() === " " || peek() === "\t")) {
-                let endOfFirstWord = current;
+                let endOfFirstWord = {
+                    position: current,
+                    column: column
+                };
 
                 // skip past any whitespace
                 let whitespace = "";
@@ -523,7 +526,8 @@ export class Lexer {
                     return;
                 } else {
                     // reset if the last word and the current word didn't form a multi-word Lexeme
-                    current = endOfFirstWord;
+                    current = endOfFirstWord.position;
+                    column = endOfFirstWord.column;
                 }
             }
```

Reply:
Thanks. That did the trick!